module/zfs/vdev_raidz.c

```diff
@@ -2744,6 +2842,175 @@ vdev_raidz_worst_error(raidz_row_t *rr)
 	return (error);
 }
 
+/*
+ * Find the median value from a set of n values
+ */
+static uint64_t
+latency_median_value(const uint64_t *data, size_t n)
+{
+	uint64_t m;
+
+	if (n % 2 == 0)
+		m = (data[(n>>1) - 1] + data[n>>1]) >> 1;
+	else
+		m = data[((n + 1) >> 1) - 1];
+
+	return (m);
+}
+
+/*
+ * Calculate the outlier fence from a set of n latency values
+ *
+ * fence = Q3 + 2 x (Q3 - Q1)
+ */
+static uint64_t
+latency_quartiles_fence(const uint64_t *data, size_t n)
+{
+	uint64_t q1, q3;
+
+	q1 = latency_median_value(&data[0], n>>1);
+	if (n % 2 == 0)
+		q3 = latency_median_value(&data[n>>1], n>>1);
+	else
+		q3 = latency_median_value(&data[(n+1) >> 1], n>>1);
+
+	uint64_t iqr = q3 - q1;
+	uint64_t fence = q3 + iqr;
```

Comment:
In comment you write `fence = Q3 + 2 x (Q3 - Q1)`,  but I don't see the `2x` here.

Reply:
Looking back across branches not sure how it mutated. I'll put it back to:
`return (q3 + ((q3 - q1) << 2));`

Reply:
Just in case, `<< 2` is `4x`, not `2x`.  But in general you don't need to be so heavy on bit shifts. Compilers can optimize those things just fine.